File: app/src/main/java/com/example/muzik/response_model/Chart.java
package com.example.muzik.response_model;

import java.util.Date;
import java.util.List;

public class Chart {
    public static class SongWithView {
        private Song song;
        private List<SongView> songViews;

        public Song getSong() {
            return song;
        }

        public List<SongView> getSongViews() {
            return songViews;
        }
    }

    public static class SongView implements Comparable<SongView> {
        private int songID;
        private Date date;
        private int views;

        public int getSongID() {
            return songID;
        }

        public Date getDate() {
            return date;
        }

        public int getViews() {
            return views;
        }

        @Override
        public int compareTo(SongView o) {
            return date.compareTo(o.date);
        }
    }
}

File: app/src/main/java/com/example/muzik/data_model/api_model/response/Chart.kt
package com.example.muzik.data_model.api_model.response

import com.example.muzik.data_model.standard_model.Song
import java.util.Date

class Chart {
    data class SongWithView(
        val song: Song,
        var songViews: List<SongView>
    )

    data class SongView(
        val songID: Long = 0,
        val date: Date,
        val views: Long = 0,
    )
}